src/System.IO.FileSystem/src/System/IO/UnixFileSystem.cs

```diff
@@ -504,6 +505,17 @@ private sealed class FileSystemEnumerable<T> : IEnumerable<T>
                         }
                         searchPattern = searchPattern.Substring(lastSlash + 1);
                     }
+
+                    // Typically we shouldn't see either of these cases
+                    if (searchPattern.IndexOfAny(s_EscapeCharacters) != -1)
```

Comment:
> Doing a manual loop, however, was nearly 5 times as fast. Switching to that.

Before you do that, is it worthwhile adding such additional code?  Will it be measurable in this context and not just in a microbenchmark?  If it's really faster, is there a fix that can be made to IndexOfAny instead?